microfast.F90

```diff
@@ -131,6 +136,9 @@ subroutine microfast(carma, cstate, iz, scale_threshold, rc)
     enddo
   enddo
 
+  call coremasscheck( carma, cstate, iz, .true.,.false.,.false., "AfterPsolve", rc )
```

Comment:
I don't think we need to check mass after psolve

Reply:
I guess you only know by testing, but I think there should be options controlled by the CARMA object to have these report. This would allow you to see where fixes are happening when you run the code, but shouldn't happen normally.

Reply:
@fvitt @cbardeen 
I assume Chuck suggest in CARMA:
if(DEBUG)
call coremasscheck( carma, cstate, iz, .true.,.true.,.true., "AfterPsolve", rc )
else
call coremasscheck( carma, cstate, iz, .true.,.false.,.false., "AfterPsolve", rc )
end if

And I suggest  delete this call after psolve.

Reply:
The ones that are fixing and not reporting, I think you always have to have, but whether they report or not should be controllable by a new flag. For exmaple

call coremasscheck( carma, cstate, iz, .true.,mass_checks_report_all,.false., "AfterPsolve", rc )

In the part of the code where you aren't fixing and just checking, I think you want to either only do those for CARMA_DEBUG or add another flag.

if (mass_checks_always) then call coremasscheck( carma, cstate, iz, .false.,true,.true., "elswhere", rc )

Reply:
@zywshoon

Reply:
done